rskj-core/src/main/java/co/rsk/peg/Federation.java

```diff
@@ -91,18 +91,7 @@ public long getCreationBlockNumber() {
         return creationBlockNumber;
     }
 
-    public Script getRedeemScript() {
-        if (redeemScript == null) {
-            redeemScript = ScriptBuilder.createRedeemScript(getNumberOfSignaturesRequired(), getBtcPublicKeys());
-        }
-
-        return redeemScript;
-    }
-
-    public Script getStandardRedeemScript() {
-        return getRedeemScript();
-    }
-
+    public abstract Script getRedeemScript();
     public Script getP2SHScript() {
         if (p2shScript == null) {
             p2shScript = ScriptBuilder.createP2SHOutputScript(getNumberOfSignaturesRequired(), getBtcPublicKeys());
```

Comment:
```suggestion
            p2shScript = ScriptBuilder.createP2SHOutputScript(getRedeemScript());
```
The P2SH script is built from the redeem script `OP_HASH160 redeemScriptHash OP_EQUAL` if I'm not mistaken. See the implementation for `createP2SHOutputScript` that receives the threshold and the public keys. It will build a default redeem script with those values. This will not work for federations that use different redeem script formats. So, it's best to pass the redeem script directly

Reply:
You are absolutely right. In my mind it was getting the redeem script directly. Thanks!